packages/team-management/src/components/TeamsAddEdit/index.tsx

```diff
@@ -90,6 +91,73 @@ export const defaultProps = {
   opensrpBaseURL: '',
 };
 
+/**
+ * function to fetch a paginated practitioner object
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param practitionersEndpoint - OpenSRP practitioners endpoint
+ * @param pageNumber - paginated page number
+ * @param pageSize - paginated page size
+ * @param errorNotification - the error message to show for a failed request
+ * @returns {Promise<Practitioner[]>} an array of practitioners
+ */
+async function fetchPractitioners(
+  opensrpBaseURL: string,
+  practitionersEndpoint: string,
+  pageNumber: number,
+  pageSize: number,
+  errorNotification: string = lang.ERROR_OCCURRED
+): Promise<Practitioner[]> {
+  const composePaginatedEndpoint = `${practitionersEndpoint}?pageNumber=${pageNumber}&pageSize=${pageSize}`;
+  const serve = new OpenSRPService(composePaginatedEndpoint, opensrpBaseURL);
+  try {
+    const practitioners: Practitioner[] = await serve.list();
+    return practitioners;
+  } catch (_) {
+    sendErrorNotification(errorNotification);
+    return [];
+  }
+}
+
+/**
+ * function to fetch paginated practitioners resource recursively
+ *
+ * @param opensrpBaseURL - OpenSRP API base URL
+ * @param practitionersCountEndpoint - OpenSRP practitioners count endpoint
+ * @param practitionersEndpoint - OpenSRP practitioners endpoint
+ * @returns {Promise<Practitioner[]>} - an array of all practitioners in a paginated endpoint
+ */
+async function fetchPractitionersRecursively(
+  opensrpBaseURL: string,
+  practitionersCountEndpoint: string,
+  practitionersEndpoint: string
+): Promise<Practitioner[]> {
+  // practitioner array to fill recursively
+  const allPractitioners: Practitioner[] = [];
+
+  // get the total number of practitioners
+  const serve = new OpenSRPService(practitionersCountEndpoint, opensrpBaseURL);
+  const practitionerCount: number = await serve.list();
+
+  const pageSize = 1000;
```

Comment:
can we make this value configurable? i.e have it as an env var

Reply:
sure

Reply:
8b692587fc9fbec8b781e755c925fb89ceb3cc5c